var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -176,9 +176,10 @@ def libs(self):
             "14",
             # C++17 is not supported by Boost < 1.63.0.
             conditional("17", when="@1.63.0:"),
-            # C++20/2a is not support by Boost < 1.73.0
+            # C++20/2a is not supported by Boost < 1.73.0
             conditional("2a", when="@1.73.0:"),
             conditional("20", when="@1.77.0:"),
+            conditional("23", when="@1.77.0:"),
```

Comment:
I think:
```suggestion
            conditional("23", when="@1.79.0:"),
```
but I will get back to you with confirmation.

Reply:
Confirmed. C++23, 2c, and 26 were added to boost/build in 4.8.2 which was part of boost 1.79.0.

Reply:
Thanks @hainest for confirming.

In that case... might as well add 26 to the list too? Compiler support is very early to say the least, but at least it'll be there ready to be used if someone feels like trying it out. Since there are currently no constraints for which _compilers_ support the flag I guess 26 (and 23) can be added without further constraints and then it's up to the user to make sure they only use 23 and 26 with compilers that support them.

Reply:
I added both 23 and 26, let me know if I should change anything

Reply:
@msimberg Might as well. I'm pretty sure there are no explicit checks for it, but might be best to give it a quick compile just to make sure there aren't any accidental breakages.

Reply:
I can confirm that the `cxxstd=23` build works fine (for a subset of libraries that I tested, but importantly `b2` accepts it from 1.79.0 onwards).

I don't have gcc or clang trunk handy to test `cxxstd=26` fully locally. It looks like for both gcc and clang, the latest released versions don't accept 26 yet. However, I can confirm that `b2` does accept the 26, the build just fails later when it comes to actually compiling something.